core/src/avm1/activation.rs

```diff
@@ -1185,59 +1185,90 @@ impl<'a, 'gc, 'gc_context> Activation<'a, 'gc, 'gc_context> {
     fn action_get_url_2(&mut self, action: GetUrl2) -> Result<FrameControl<'gc>, Error<'gc>> {
         // TODO: Support `LoadVariablesFlag`, `LoadTargetFlag`
         // TODO: What happens if there's only one string?
-        let target = self.context.avm1.pop();
+        let target_val = self.context.avm1.pop();
+        let target = target_val.coerce_to_string(self)?;
         let url_val = self.context.avm1.pop();
         let url = url_val.coerce_to_string(self)?;
 
         if let Some(fscommand) = fscommand::parse(&url) {
-            let fsargs = target.coerce_to_string(self)?;
-            fscommand::handle(fscommand, &fsargs, self)?;
+            // `target` = fscommand arguments!
+            fscommand::handle(fscommand, &target, self)?;
             return Ok(FrameControl::Continue);
         }
 
-        let window_target = target.coerce_to_string(self)?;
-        let clip_target: Option<DisplayObject<'gc>> = if action.is_target_sprite() {
-            if let Value::Object(target) = target {
+        // TODO: Use `StageObject::get_level_by_path`.
+        let level_target = if target.starts_with(WStr::from_units(b"_level")) && target.len() >= 6 {
+            match target[6..].parse::<f64>() {
+                Ok(level_id) => level_id as i32,
+                Err(_) => {
+                    if target.len() == 6 {
+                        0
+                    } else {
+                        -1
+                    }
+                }
+            }
+        } else {
+            -1
+        };
```

Comment:
I think it's worthwhile to resolve the TODO in this PR and switch this over to using `StageObject::get_level_by_path`, because it will clean this up and also handle these weirdo cases:
`loadMovieNum("child.swf", 4294967297); // wraps to _level1`
`loadMovie("child.swf", "_flash1"); // way old synonym for _level1`

Possibly `StageObject::get_level_by_path` could be moved somewhere more appropriate as a utility function.

Reply:
Thanks for the review.

This TODO is actually more recent than this PR (cc. #6434), that's why I didn't really consider it here.

I've made an alternative version that could eventually supersede this PR but more changes are implied:
https://github.com/Toad06/ruffle/commit/304ae9f77c7823dbc92d638eda64687fdda23d97

To sum up what this PR does:
- It uses `StageObject::parse_level_id`, as `StageObject::get_level_by_path` won't resolve a level that doesn't exist.
- `StageObject::parse_level_id` is changed to parse the whole string, not just the part after "_level". The function also has a new argument `strict_parse`, as things like `_level1ABC` only resolves to `_level1` in `action_get_url_2` if this level already exists.

All of this is still very experimental to me.